invoice/src/main/java/org/killbill/billing/invoice/model/InvoiceItemFactory.java

```diff
@@ -141,7 +141,14 @@ public static InvoiceItem fromModelDaoWithCatalog(final InvoiceItemModelDao invo
             if (computePrettyName) {
 
                 if (planName != null) {
-                    final Plan plan = catalog.findPlan(planName, createdDate);
+
+                    // We don't know the creation date of the said subscription, all we know is the time
+                    // of the current event. By specifying KILLBILL_GENESIS, we should at least find a Plan
+                    // and most likely the correct one since we start from the transitionDate and move backward
+                    // Worst case: This is the wrong plan and the prettyName was updated across version, oh well..
+                    //
+                    final DateTime KILLBILL_GENESIS = new DateTime(2019, 10, 28, 0, 0);
```

Comment:
⚠️ Confused by the chosen date? Should it be 2011 instead?

Also,  this should probably be `static final`, and with a timezone.

Reply:
Yes, you are right, this is a typo, was supposed to be 2011, will fix. I can add the timezone, but to match the exact date of our first commit `Fri Oct 28 15:06:57 2011 -0700`. Not sure this part really matters

Reply:
I meant to say that you should specify `UTC` for the date (`CatalogDateHelper` works in UTC).